### ISSUE TITLE
I propose we support automatic creation of FormattableStrings

### ISSUE THREAD

**Issue by Robert923** [fsharp/fslang-suggestions#669]
# Title of Suggestion

I propose we support automatic creation of FormattableStrings from strings where the parameter type on the function being called is a FormattableString.  For example, when using [Entity Framework 2.0](https://blogs.msdn.microsoft.com/dotnet/2017/06/28/announcing-ef-core-2-0-preview-2/) the FromSql and ExecuteSqlCommand now require a FormattableString.

In C# or VB.NET this is quite elegant and helps reduce errors.

However the existing approach to this problem in F# is that the developer must manually open System.Runtime.CompilerServices and use the FormattableStringFactory.Create() method:
```
            let sql = FormattableStringFactory.Create("DELETE FROM mytable WHERE country = '**'")
            dbContext.Database.ExecuteSqlCommand(sql) |> ignore
```

The advantages of making this adjustment to F# are that modern libraries such as EF become more usable from F#.

I'm not sure there are any disadvantages to making this adjustment to F#.

## Extra information

Estimated cost S 

Related suggestions: It may be better (but larger) to add more complete formatted string support.

## Affidavit (please submit!)

Please tick this by placing a cross in the box:
* [x ] This is not a question (e.g. like one you might ask on [stackoverflow](http://stackoverflow.com)) and I have searched stackoverflow for discussions of this issue
* [ x] I have [searched both open and closed suggestions on this site](http://github.com/fsharp/fslang-suggestions/issues) and believe this is not a duplicate
* [x ] This is not something which has obviously "already been decided" in previous versions of F#.  If you're questioning a fundamental design decision that has obviously already been taken (e.g. "Make F# untyped") then please don't submit it.

Please tick all that apply:
* [ ] This is not a breaking change to the F# language design
* [ x ] I or my company would be willing to help implement and/or test this

**Comment by rmunn** [comment]
Instead of automagically converting strings to FormattableStrings based on function types, which goes against the "no auto-typecasting" design philosophy that F# has always had (you can't even auto-convert an int32 to an int64), I'd suggest that something like https://github.com/fsharp/fslang-design/issues/6#issuecomment-343472521 would be a better idea: a string prefix, maybe `f`, that produces a FormattableString instead of a "regular" string literal. E.g., `"foo"` is a string, and `f"foo"` is a FormattableString.

**Comment by Robert923** [comment]
Agreed, except that there's precipitant for format strings - [`printf`](https://msdn.microsoft.com/en-us/visualfsharpdocs/conceptual/core.printf-module-%5bfsharp%5d) automatically converts a string into TextWriterFormat, performing type checking at the same time.

It's been my experience working with EF Core 2 and ASP.NET Core 2 that the capabilities these great platforms provide are more and more built specificaly for C#, and there is a lot of friction when using them from F#.   I was hoping we could reduce the friction slightly in a case similar to printf

**Comment by dsyme** [comment]
AFAIU this is effectively handled  by string interpolation.
